### PR TITLE
MEN-4423: Acceptance tests for mender-configure files

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: tests/test_rootfs.py [hunk 1]
```diff
@@ -54,6 +54,30 @@ def verify_fstab(data):
...
+        ).decode()
+        file_found = any(
+            [
+                line.split("/")[5] == filename
+                for line in output.split("\n")
+                if len(line) > 0
+            ]
+        )
+        assert (
+            expect_to_exist and file_found or (not expect_to_exist and not file_found)
```
Comment (kacf) [inline tests/test_rootfs.py:70]: Nitpick: I would put parentheses around both ands, just for clarity.

File: tests/test_rootfs.py [hunk 2]
```diff
@@ -54,6 +54,30 @@ def verify_fstab(data):
             )
             occurred[cols[1]] = True
 
+    @staticmethod
+    def verify_file_exists(tmpdir, rootfs, path, filename, expect_to_exist=True):
```
Comment (kacf) [inline tests/test_rootfs.py:58]: Nice cleanup!

File: tests/test_rootfs.py [hunk 3]
```diff
@@ -204,6 +192,54 @@ def test_expected_files_ext234_mender_connect(
...
+            )
+
+            # Check whether mender-inventory-mender-configure is executable
+            if expect_installed:
+                self.verify_file_executable(
+                    tmpdir,
+                    latest_rootfs,
+                    "/usr/share/mender/inventory",
+                    "mender-inventory-mender-configure",
+                )
```
Comment (kacf) [inline tests/test_rootfs.py:241]: We should check (and implement) two more things here: That `/var/lib/mender-configure` is a symlink which points to `/data/mender-configure`, and that the latter exists.

**Review by kacf — CHANGES_REQUESTED** [review]

File: tests/test_rootfs.py (hunk 3 above)
Comment (kacf) [reply tests/test_rootfs.py:241]: I don't see a test for either of these two things... Verifying `/data/mender-configure` needs to be in `test_part_image.py`, I suppose.

Comment (lluiscampos) [reply tests/test_rootfs.py:241]: Sorry! You are right.

I implemented it but forgot to test. I'll extend the test now and create a new meta-mender PR to verify.

Comment (lluiscampos) [reply tests/test_rootfs.py:241]: > I don't see a test for either of these two things... [,,,]

Well, we check that `/var/lib/mender-configure` exists  only, not that it is a symlink. See https://github.com/mendersoftware/mender-image-tests/pull/39/files#diff-61f29111cea287297ab544f795cf48c3a8f83c2b5a24628fef969b8e65215772R243

Comment (kacf) [reply tests/test_rootfs.py:241]: I know. But it's important that it's a symlink! Important enough that we should test it I think. :-)

**Comment by lluiscampos** [comment]
@kacf See latest commit:

```MEN-4423: Extend tests to verify /var/lib and /data contents
    
    Namely:
    * Extend test_expected_files_ext234_mender_configure to verify symlink
      from /var/lib/mender-configure to /data/mender-configure
    * Extend test_expected_files_ext234 to similarly vereify symlink from
      /var/lib/mender to /data/mender (just to be consistent)
    * New test test_expected_data_dirs to verify dirs in /data partition
```

I am launching ASAP a new PR on meta-mender to run these tests.

File: tests/test_part_image.py [hunk 4]
```diff
@@ -379,6 +379,38 @@ def test_split_mender_conf(
...
+                extract_partition(latest_part_image, data_part_number)
+
+                output = subprocess.check_output(
+                    ["debugfs", "-R", "ls -l -p /", f"img{data_part_number}.fs"]
+                ).decode()
+                nodes = [
+                    line.split("/")[5] for line in output.split("\n") if len(line) > 0
+                ]
+
+                # Expect /var/lib/mender to exist always
```
Comment (kacf) [inline tests/test_part_image.py:401]: Nitpick: `Expect /data/mender to exist always`

**Review by kacf — APPROVED** [review]
:+1:

**Pull request merged**